docs/sidechain/spec.md

```diff
@@ -0,0 +1,945 @@
+   at the time the quorum is reached is controlling. When the quorum is reach,
+   the signing keys will be checked against the current signers list, and if a
+   collected signature's key is no longer on that list it is removed and
+   signatures will continue to be collected.
+   
+5) On the destination chain, the the owner of the "cross chain sequence number"
+   (see 1) can submit a "cross chain claim" transaction that includes the "cross
+   chain sequence number", the sidechain, and the destination. The "cross chain
+   sequence number" object must exist and must have already collected enough
+   signatures from the witness servers for this to succeed. On success, a
+   payment is made from the door account to the destination and the "cross chain
+   sequence number" is deleted. A "cross chain claim" transaction can only
+   succeed once, as the "cross chain sequence number" for that transaction can
+   only be created once. In case of error, the funds can be sent to an alternate
+   account and eventually returned to the initiating account. Note that this
+   transaction is only used if the optional destination account is not specified
+   in step (2) or there is an error when sending funds to that destination
+   account.
+
+6) When a witness server sees funds being claimed on the destination chain, it
+   submits a transaction on the source chain allowing the signature rewards to
+   be collected. It collects these signatures on the `XChainTransferRewardState`
+   created in step (2). The number of rewards will always equal the number of
+   signatures needed to reach a quorum. When a quorum of these signatures have
+   been collected, the rewards will be transferred to the destination addresses
+   specified in the signatures. (Note: the witness servers specify where the rewards
+   for its signature goes, this is not specified on the sidechain ledger
+   object). The `XChainTransferRewardState` is destroyed when the reward pool is
+   distributed.
+   
+The cross-chain transfer is now complete. Note that the transactions sent by the
+witness servers that add their signatures may send the signatures in a batch.
+
+## Supporting transactions overview
+
+In addition to the transactions used in a cross-chain transfer (described
+above), there are new transactions for creating a sidechain, changing sidechain
+parameters, and for using a cross-chain transfer to create a new account on the
+destination chain.
+
+The `SidechainCreate` transaction adds a sidechain ledger object to the account.
+This contains the two door accounts, the asset type that will be put into trust
+on the mainchain, the wrapped asset type on the sidechain, and the reward amount
+in XRP per signature. Optional, the minimum amount of XRP needed for an account
+create transaction for the mainchain, the minimum amount of XRP needed for an
+account create transaction for the sidechain may be specified. Currently, this
```

Comment:
> Optional, the minimum amount of XRP needed for an account create transaction for the mainchain, the minimum amount of XRP needed for an account create transaction for the sidechain may be specified.

Wouldn't you only need the account create reserve for the opposite chain?

Reply:
Does it have to specify the account minimum for both chains, or just the other chain? That was the confusion I had.

Reply:
Oh, I see. Yes, I was planning on keeping amounts for both chains. I should only need it for the "other chain" tho.

Reply:
Yes, I was planning on keeping this for both chains. I was thinking that's it's nice that we use the exact same sidechain parameters when creating on the sidechain object on both chains. But you're right, we only need the minimum for the "other" chain.

Reply:
It will only keep the amount needed for the "other chain".